adapters/terraformcloud/membership/membership.go

```diff
@@ -0,0 +1,204 @@
+/*
+Package user synchronises members in a Terraform Cloud organisation.
+
+# Requirements
+
+In order to synchronise with Terraform cloud, you will need an Organization API token:
+https://developer.hashicorp.com/terraform/cloud-docs/users-teams-organizations/api-tokens#organization-api-tokens
+
+# Examples
+
+See [New] and [Init].
+*/
+package membership
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/hashicorp/go-tfe"
+
+	gosync "github.com/ovotech/go-sync"
+)
+
+// Token sets the authentication token for Terraform Cloud.
+const Token gosync.ConfigKey = "terraform_cloud_token"
+
+// Organisation sets the Terraform Cloud organisation.
+const Organisation gosync.ConfigKey = "terraform_cloud_organisation"
+
+var (
+	_ gosync.Adapter = &Membership{} // Ensure [membership.Membership] fully satisfies the [gosync.Adapter] interface.
+	_ gosync.InitFn  = Init          // Ensure the [membership.Init] function fully satisfies the [gosync.InitFn] type.
+)
```

Comment:
I'd normally expect to see interface checks like this in a `*_test.go` file instead of the internal package code; I think they just make more sense living in a test file.
Both of those types being checked are exported, so you wouldn't even need to cheat with an internal test file. 😬

Reply:
That makes sense.  I'll update this.  Thanks

Reply:
✅ Fixed by https://github.com/ovotech/go-sync/pull/232/commits/aee24725fa2decc53699f1f283483c1ccbd25d73